gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposal/acceptProposal.ts

```diff
@@ -1,47 +1,158 @@
+import { PublicKey } from '@solana/web3.js'
+import { TOKEN_PROGRAM_ID } from '@solana/spl-token'
 import { utils } from '@project-serum/anchor'
-import { ORACLES_MAX_LENGTH } from '../../../../lib/constants'
 import { CONTRACT_LIST, getContract } from '../../../../lib/contracts'
+import ProposeOffchainConfig, { OffchainConfig } from '../proposeOffchainConfig'
 import { getRDD } from '../../../../lib/rdd'
-import { makeTx } from '../../../../lib/utils'
+import { serializeOffchainConfig } from '../../../../lib/encoding'
 
 type Input = {
-  digest: Buffer
+  version: number
+  f: number
+  tokenMint: string
+  oracles: {
+    transmitter: string
+    signer: string
+    payee: string
+  }[]
+  offchainConfig: OffchainConfig
+  randomSecret: string
+}
+
+type DigestInput = {
+  version: BN
+  f: BN
+  tokenMint: PublicKey
+  oracles: {
+    transmitter: PublicKey
+    signer: Buffer
+    payee: PublicKey
+  }[]
+  offchainConfig: Buffer
+}
+
+type Proposal = {
+  owner: PublicKey
+  state: number
+  f: number
+  tokenMint: PublicKey
+  oracles: {
+    xs: {
+      transmitter: PublicKey
+      signer: {
+        key: Buffer
+      }
+      payee: PublicKey
+    }[]
+    len: number
+  }
+  offchainConfig: {
+    version: number
+    xs: Buffer
+    len: number
+  }
 }
 export default class AcceptProposal extends SolanaCommand {
   static id = 'ocr2:accept_proposal'
   static category = CONTRACT_LIST.OCR_2
 
   static examples = [
-    'yarn gauntlet ocr2:accept_proposal --network=devnet --state=EPRYwrb1Dwi8VT5SutS4vYNdF8HqvE7QwvqeCCwHdVLC',
+    'yarn gauntlet ocr2:accept_proposal --network=devnet --state=<STATE_ADDRESS> --proposalId=<PROPOSAL_ID> --rdd=<PATH_TO_RDD>',
   ]
 
   makeInput = (userInput): Input => {
     if (userInput) return userInput as Input
-    // const rdd = getRDD(this.flags.rdd)
-    // const aggregator = rdd.contracts[this.flags.state]
-    // const aggregatorOperators: any[] = aggregator.oracles.map((o) => rdd.operators[o.operator])
-    // const oracles = aggregatorOperators.map((operator) => ({
-    //   // Same here
-    //   transmitter: operator.ocrNodeAddress[0],
-    //   signer: operator.ocr2OnchainPublicKey[0].replace('ocr2on_solana_', ''),
-    // }))
-    const digest = Buffer.alloc(32) // TODO
 
+    const rdd = getRDD(this.flags.rdd)
+    const aggregator = rdd.contracts[this.flags.state]
+    const _toHex = (a: string) => Buffer.from(a, 'hex')
+    const aggregatorOperators: any[] = aggregator.oracles.map((o) => rdd.operators[o.operator])
+    const oracles = aggregatorOperators
+      .map((operator) => ({
+        transmitter: operator.ocrNodeAddress[0],
+        signer: operator.ocr2OnchainPublicKey[0].replace('ocr2on_solana_', ''),
+        payee: operator.adminAddress,
+      }))
+      .sort((a, b) => Buffer.compare(_toHex(a.signer), _toHex(b.signer)))
+    const offchainConfig = ProposeOffchainConfig.makeInputFromRDD(rdd, this.flags.state)
+    const tokenMint = process.env.LINK!
```

Comment:
Rather than using a `LINK` env var you can fetch the `tokenMint` off the proposal account